core/primitives/src/hash.rs

```diff
@@ -10,6 +10,7 @@ pub struct CryptoHash(pub Digest);
 
 impl CryptoHash {
     pub fn new(data: &[u8]) -> Self {
+        assert_eq!(data.len(), 32);
```

Comment:
this is not rust-like.. this should return a `Result<CryptoHash, ?>` and return an `Err` rather than panic

Reply:
that will be a lot of pain. removed assert but for debugging this assert is nice